src/conn.rs

```diff
@@ -184,6 +184,11 @@ pub struct Connection<const N: usize, P: ConnectionPeer> {
     pending_writes: VecDeque<QueuedWrite>,
     writable: Notify,
     latest_timeout: Option<Instant>,
+    // Save the first STATE packet returned to an incoming connection request.
+    // We resend the exact packet in case the first response to the SYN packet is lost,
+    // because regenerating the STATE packet gives us the wrong sequence number, and causes the
+    // connecting peer to ignore earlier data sent to them.
+    syn_ack: Option<Packet>,
```

Comment:
```suggestion
    /// Save the first STATE packet returned to an incoming connection request.
    /// We resend the exact packet in case the first response to the SYN packet is lost,
    /// because regenerating the STATE packet gives us the wrong sequence number, and causes the
    /// connecting peer to ignore earlier data sent to them.
    syn_ack: Option<Packet>,
```

This should be a doc comment

Reply:
The thought had occurred to me as I was writing it, and then I thought maybe a private variable shouldn't be?

Just now, I searched around a bit and found [this](https://github.com/rust-lang/rust-by-example/issues/1272), which prefers doc comments for private variables. There is no explanation why, so it's entirely an appeal to authority, which is among my least favorite. But I don't have a strong argument the other way. Happy to switch 👌🏻

Reply:
For what it's worth, the generated docs seem to skip over private fields. So I'm not convinced that making this a doc-comment has any value. See https://docs.rs/utp-rs/0.1.0-alpha.17/utp_rs/conn/struct.Connection.html :

> pub struct Connection<const N: [usize](https://doc.rust-lang.org/nightly/std/primitive.usize.html), P: > [ConnectionPeer](https://docs.rs/utp-rs/0.1.0-alpha.17/utp_rs/peer/trait.ConnectionPeer.html)> { /* private fields */ }

Reply:
Code editors will display the doc comments for a private variable when you are trying to use it or hover over it

![image](https://github.com/user-attachments/assets/e08eb4e4-1454-4d15-8214-4bfd15e01fe0)

So there is value when coding

Reply:
![image](https://github.com/user-attachments/assets/751170ff-a730-424e-87f1-3d4682ae3d18)

Reply:
![image](https://github.com/user-attachments/assets/1541a794-35d0-4a47-a06b-d190065d4398)

See it is nice :)

Reply:
Yup, that's a good reason! 👌🏻